### PR TITLE
multi-data for pf-utilization-bar-chart

### PR DESCRIPTION
**Pull request by kamleshverma1**
multi-data with multi-color support for single utilization bar chart

Signed-off-by: kamlesh <kaverma@gmail.com>

### CODE REVIEW

**Comment by dtaylor113** [comment]
Hi, cloned, built, viewed ng-docs, and reviewed the code.  While the coding looks solid, we are not sure if this new feature breaks the intention of the utilization bar whose main purpose is to be able to quickly view the thresholds of the various bars (Error, Warning, Ok).  The new sub-item/divisions of the Used bar doesn't appear to use the thresholds for coloring.  We are currently discussing whether this new type of utilization bar should live here, or perhaps be a new angular-patternfly directive.

File: test/charts/utilization-bar/utilization-bar.spec.js [hunk 1]
```diff
@@ -32,6 +36,18 @@ describe('Directive: pfUtilizationBarChart', function() {
...
   };
 
+  it("should set value of data.subdata length to be 3", function() {
+    $scope.layout.type = 'multidata';
+    $scope.data = {
+      'total': '100',
+      'subdata' : [ { "used" : 45 , "color" : "#00558a" , "subtitle" : "Object" },
+                    { "used" : 15 , "color" : "#0071a6" , "subtitle" : "Block" },
+                    { "used" :  5 , "color" : "#00a8e1" , "subtitle" : "OpenStack" }]
+    };
```
Comment (jeff-phillips-18) [inline test/charts/utilization-bar/utilization-bar.spec.js]: Please choose colors from the Patternfly color palette -  [Color Palette](https://www.patternfly.org/styles/color-palette/)

File: styles/angular-patternfly.css [hunk 2]
```diff
@@ -344,6 +344,27 @@
...
+  margin-top: -19px;
+}
+
+.multidata-progress-description {
+  overflow: unset !important;
+  white-space: normal !important;
+  margin-bottom: 0px !important;
+  text-align: -moz-left !important;
+}
+
```
Comment (jeff-phillips-18) [inline styles/angular-patternfly.css]: Please put these selectors under the utilization-bar-chart-pf selector.

File: src/charts/utilization-bar/utilization-bar-chart.directive.js [hunk 3]
```diff
@@ -107,6 +110,18 @@
       'type': 'inline'
     };
 
+    $scope.title6 = 'Utilization by storage type';
+    $scope.data6 = {
+      'total': '100',
+      'subdata' : [ { "used" : 45 , "color" : "#00558a" , "subtitle" : "Object" },
+                    { "used" : 15 , "color" : "#0071a6" , "subtitle" : "Block" },
+                    { "used" :  5 , "color" : "#00a8e1" , "subtitle" : "OpenStack" }]
```
Comment (jeff-phillips-18) [inline src/charts/utilization-bar/utilization-bar-chart.directive.js]: Please choose colors from the Patternfly color palette - [Color Palette](https://www.patternfly.org/styles/color-palette/)

**Comment by dtaylor113** [comment]
Hi, this enhancement deviates enough from the utilization bar pattern to warrant a new patternfly component.  This work is being tracked via: https://patternfly.atlassian.net/browse/PTNFLY-1229
I new PR will be created once the new component becomes available.

**Pull request closed without merging**